lib/travis/build/script/go.rb

```diff
@@ -110,10 +110,10 @@ def uses_make
           
           # see https://golang.org/doc/go1.6#go_command
           def uses_15_vendoring
-            if (go_version == 'go1' || comparable_go_version < Gem::Version.new('1.5'))
+            if (go_version == 'go1' || (go_version != 'tip' && comparable_go_version < Gem::Version.new('1.5')))
```

Comment:
Also, `go_version != 'tip' && comparable_go_version < Gem::Version.new('1.5')` appears a few times, and I think it should also be extracted into a method.